featran-jmh/src/test/scala/com/spotify/featran/jmh/HotEncoderTransformersBenchmark.scala

```diff
@@ -0,0 +1,234 @@
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional1000Values)
+
+  @Benchmark
+  def nHotRequiredWithListMatch100000Values: Unit =
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional100000Values)
+
+  @Benchmark
+  def nHotRequiredWithListMatch100000ValuesWR: Unit =
+    nHotRequiredWithListMatchPrepare(arbitraryListOptional100000ValuesWR)
+
+  //
+  // ONE HOT - optional
+  //
+  @Benchmark
+  def oneHotOptionalEncoder: Unit = oneHotOptional(arbitraryString)
+
+  @Benchmark
+  def oneHotOptionalEncoder1000Values: Unit =
+    oneHotOptional(arbitraryString1000Values)
+
+  @Benchmark
+  def oneHotOptionalEncoder100000Values: Unit =
+    oneHotOptional(arbitraryString100000Values)
+
+  @Benchmark
+  def oneHotOptionalEncoder100000ValuesWR: Unit =
+    oneHotOptional(arbitraryString100000ValuesWR)
+
+  //
+  // ONE HOT - required
+  //
+  @Benchmark
+  def oneHotRequiredEncoder: Unit = oneHotRequired(arbitraryString)
+
+  @Benchmark
+  def oneHotRequiredEncoder1000Values: Unit =
+    oneHotRequired(arbitraryString1000Values)
+
+  @Benchmark
+  def oneHotRequiredEncoder100000Values: Unit =
+    oneHotRequired(arbitraryString100000Values)
+
+  @Benchmark
+  def oneHotRequiredEncoder100000ValuesWR: Unit =
+    oneHotRequired(arbitraryString100000ValuesWR)
+
+  def nHotOptional(
+      input: List[List[String]]): FeatureExtractor[List, List[String]] = {
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .optional {
+        case List("") => None
+        case i        => Option(i)
+      }(NHotEncoder("n_hot"))
+    featureSpec.extract(input)
+  }
+
+  def nHotRequired(
+      input: List[List[String]]): FeatureExtractor[List, List[String]] = {
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .required(identity)(NHotEncoder("n_hot"))
+    featureSpec.extract(input)
+  }
+
+  def nHotRequiredWithListMatchPrepare(
+      input: List[Option[String]]): FeatureExtractor[List, List[String]] = {
+    val listInput = input.map {
+      case Some(i) => List(i)
+      case _       => List()
+    }
+    val featureSpec = FeatureSpec
+      .of[List[String]]
+      .required(identity)(NHotEncoder("n_hot"))
+    featureSpec.extract(listInput)
+  }
+
+  def oneHotOptional(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .optional {
+        case "" => None
+        case f  => Option(f)
+      }(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  def oneHotRequired(input: List[String]): FeatureExtractor[List, String] = {
+    val featureSpec = FeatureSpec
+      .of[String]
+      .required(identity)(OneHotEncoder("one_hot"))
+    featureSpec.extract(input)
+  }
+
+  // Lists of Lists for N-Hot
+  private implicit val arbitraryListString: List[List[String]] = List(
+    List(thisString) ++ Seq(""))
+  private implicit val arbitraryListString1000Values: List[List[String]] = List(
+    sample(data, 0.01).toList)
```

Comment:
This may not return exactly 10000 values right?
I'd do something like this:

```
val vocab = (1 to 100).map("label" + _)
val s1000 = Seq.fill(1000)(vocab(rng.nextInt(100)) // pick 1
```